packages/agora-rtc-react/src/hooks/useRemoteAudioTracks.ts

```diff
@@ -0,0 +1,119 @@
+import type { IAgoraRTCClient, IAgoraRTCRemoteUser, IRemoteAudioTrack } from "agora-rtc-sdk-ng";
+
+import { useCallback, useEffect, useState } from "react";
+import { joinDisposers } from "../utils";
+import { listen } from "../listen";
+import { useRTCClient } from "./context";
+import { useIsConnected } from "./client";
+
+/**
+ * Auto-subscribe and get remote user audio track.
+ * Unsubscribe track on unmount.
+ */
+export function useRemoteAudioTracks(
+  users: IAgoraRTCRemoteUser[] | undefined,
+  client?: IAgoraRTCClient | null,
+): {
+  audioTracks: IRemoteAudioTrack[];
+  ready: boolean;
+  error: unknown;
+} {
+  const resolvedClient = useRTCClient(client);
+  const [tracks, setTracks] = useState<IRemoteAudioTrack[]>([]);
+  const [ready, setReady] = useState<boolean>(false);
+  const [error, setError] = useState<unknown>();
+  const isConnected = useIsConnected();
+
+  const subscribe = useCallback(
+    async (user: IAgoraRTCRemoteUser) => {
+      try {
+        setReady(false);
+        if (!user.audioTrack && resolvedClient.remoteUsers.some(({ uid }) => user.uid === uid)) {
+          await resolvedClient.subscribe(user, "audio");
+          // when HMR track will be changed, need update tracks
+          let nextTracks = [...tracks];
+
+          if (user.audioTrack && !tracks.some(track => track.getUserId() === user.uid)) {
+            nextTracks = [...tracks, user.audioTrack];
+          }
+
+          // when HMR track will be changed, need update tracks
+          nextTracks = nextTracks.map(track => {
+            if (
+              user.audioTrack &&
+              track.getUserId() === user.uid &&
+              track.getTrackId() !== user.audioTrack.getTrackId()
+            ) {
+              return user.audioTrack;
+            } else {
+              return track;
+            }
+          });
+          setTracks(nextTracks);
```

Comment:
Since this `setState` is called after an async job `client.subscribe()`, it is possible that at this time the component is unmounted, then React will print errors in the console to ask you to not do this as there could be memory leaks. You can use the `isMounted` trick like other hooks in this repo to escape from that.